src/components/databaseUi/DocStoreUI.tsx

```diff
@@ -0,0 +1,187 @@
+import DatabaseUIProvider from "../../providers/DatabaseUIProvider";
+import React, { useState } from 'react';
+import ToolbarStyle from '../viewDatabase/Sidebar.module.css';
+import { Store } from "orbit-db-store";
+import AceEditor from "react-ace";
+import Popup from "reactjs-popup";
+import "ace-builds/src-noconflict/mode-jsx";
+import ReactTable from "react-table";
+
+const languages = [
+  "javascript",
+  "json"
+];
+
+const themes = [
+  "monokai",
+  "github",
+  "tomorrow",
+  "kuroir",
+  "twilight",
+  "xcode",
+  "textmate",
+  "solarized_dark",
+  "solarized_light",
+  "terminal"
+];
+
+languages.forEach(lang => {
+  require(`ace-builds/src-noconflict/mode-${lang}`);
+  require(`ace-builds/src-noconflict/snippets/${lang}`);
+});
+
+themes.forEach(theme => require(`ace-builds/src-noconflict/theme-${theme}`));
+
+export default class DocStoreUI implements DatabaseUIProvider {
+
+  getSidebar: React.FC<Store> = ({ store }) => {
+    const [editorStatus, setEditorStatus] = useState({data: "{}", open: false});
+    const [input, setInput] = useState("");
+    const [errorStatus, setErrorStatus] = useState({error: false, msg:""});
+    const [activeField, setActiveField] = useState("");
+
+    async function submitDocument(e) {
+      e.preventDefault();
+      try {
+        await store.put(JSON.parse(editorStatus.data));
+        setEditorStatus({...editorStatus, open: false});
+      } catch (error) {
+        setEditorStatus({...editorStatus, open: true});
+        setErrorStatus({error: true, msg: error});
+      }
+    }
+
+    const oneInputField = (opName, op) => {
+      return () => (<div className={ToolbarStyle.inputFieldContainer}>
+        <form onSubmit={(e) => {e.preventDefault(); op()}}>
+          <div className={ToolbarStyle.inputFieldRow}>
+            <label>Key: </label>
+            <input
+              className={ToolbarStyle.inputField}
+              type="text"
+              onChange={(e) => setInput(e.target.value)}
+            />
+          </div>
+          <div style={{display: 'flex', flexDirection: 'row', justifyContent: 'flex-end', marginTop: '0.5vh'}}>
+            <input type="submit" value={opName} className={ToolbarStyle.inputFieldSubmit}/>
+          </div>
+        </form>
+      </div>);
+    };
+
+    const addEditorField = (opName, op) => {
+      return () => (<div className={ToolbarStyle.inputFieldContainer}>
+        <form onSubmit= {op}>
```

Comment:
e.preventDefault() might be needed cause form submits traditionally cause page refresh, which is not what we want

Reply:
it has, i give it submit document which has an e.preventdefault